### PR TITLE
Latejoinfix

### PR DESCRIPTION
**Pull request by Undid-Iridium**
for @louis1706

### CODE REVIEW

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="FixLateJoin.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+using MEC;
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:8]:
```suggestion

```

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+namespace Exiled.Events.Patches.Generic
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Exiled.API.Features.Pools;
+    using GameCore;
+    using HarmonyLib;
+    using InventorySystem.Items.Firearms.Modules;
+    using Mirror;
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:19]:
```suggestion
    using MEC;
    using Mirror;
```

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                hub.roleManager.ServerSetRole(RoleTypeId.Spectator, RoleChangeReason.RoundStart);
+                Log.Debug("2.5");
+                Timing.CallDelayed(.5f, () =>
+                {
+                    Log.Debug("Delayed spawnlate.");
+                    HumanSpawner.SpawnLate(hub);
+                });
+                Log.Debug("3");
+            }
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:92]:
```suggestion
            }
            
```

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+            for (var z = 0; z < newInstructions.Count; z++)
+                yield return newInstructions[z];
+
+            ListPool<CodeInstruction>.Pool.Return(newInstructions);
+        }
+
+        private static void ProcessJoin(ReferenceHub hub, ClientInstanceMode clientInstance)
+        {
+            Log.Debug("1");
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:67]: Debug log go brrr

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            if (!NetworkServer.active || !RoleAssigner.CheckPlayer(hub) || !RoleAssigner._spawned)
+            {
+                Log.Debug("1.5");
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:70]: same as above

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                return;
+            }
+
+            if (RoleAssigner.LateJoinTimer.Elapsed.TotalSeconds >
+                ConfigFile.ServerConfig.GetFloat("late_join_time"))
+            {
+                Log.Debug("2");
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:77]: same as above

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+                hub.roleManager.ServerSetRole(RoleTypeId.Spectator, RoleChangeReason.RoundStart);
+            }
+            else
+            {
+                hub.roleManager.ServerSetRole(RoleTypeId.Spectator, RoleChangeReason.RoundStart);
+                Log.Debug("2.5");
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:83]: same as above

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                Timing.CallDelayed(.5f, () =>
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:84]: same as above

File: Exiled.Events/Patches/Generic/FixLateJoin.cs [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+                {
+                    Log.Debug("Delayed spawnlate.");
+                    HumanSpawner.SpawnLate(hub);
+                });
+                Log.Debug("3");
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:89]: same as above

**Review by Misaka-ZeroTwo — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Generic/FixLateJoin.cs (hunk 4 above)
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:67]:
```suggestion

```

File: Exiled.Events/Patches/Generic/FixLateJoin.cs (hunk 5 above)
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:70]:
```suggestion

```

File: Exiled.Events/Patches/Generic/FixLateJoin.cs (hunk 6 above)
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:77]:
```suggestion
```

File: Exiled.Events/Patches/Generic/FixLateJoin.cs (hunk 7 above)
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:83]:
```suggestion
```

File: Exiled.Events/Patches/Generic/FixLateJoin.cs (hunk 8 above)
Comment (Misaka-ZeroTwo) [inline Exiled.Events/Patches/Generic/FixLateJoin.cs:84]:
```suggestion
```

File: Exiled.Events/EventArgs/Map/ExplodingGrenadeEventArgs.cs [hunk 10]
```diff
@@ -64,15 +64,16 @@ public ExplodingGrenadeEventArgs(Player thrower, Vector3 position, EffectGrenade
         /// <param name="grenade">
         ///     <inheritdoc cref="Projectile" />
         /// </param>
-        /// <param name="players">
-        ///     <inheritdoc cref="TargetsToAffect" />
+        /// <param name="isAllowed">
+        ///     <inheritdoc cref="IsAllowed" />
         /// </param>
-        public ExplodingGrenadeEventArgs(Player thrower, EffectGrenade grenade, List<Player> players)
+        public ExplodingGrenadeEventArgs(Player thrower, EffectGrenade grenade, bool isAllowed = true)
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Map/ExplodingGrenadeEventArgs.cs:70]: breaking change?

Comment (louis1706) [reply Exiled.Events/EventArgs/Map/ExplodingGrenadeEventArgs.cs:70]: Nope

**Comment by Undid-Iridium** [comment]
NOT TESTED.

**Comment by Undid-Iridium** [comment]
This didn't fix anything, closing

**Pull request closed without merging**